client/src/features/Home/RoomNameForm.tsx

```diff
@@ -77,16 +84,28 @@ export const RoomNameForm: React.FC<Props> = ({ open, handleClose, youtubeUrl })
         <Controller
           control={roomNameForm.control}
           name="roomName"
-          render={({ field, fieldState: { error } }) => (
-            <TextField
-              label="Your room name"
-              error={!!error}
-              helperText={error?.message}
-              fullWidth
-              {...field}
-              autoFocus
-            />
-          )}
+          render={({ field: { value, onChange }, fieldState: { error } }) => {
+            return (
+              <>
+                <TextField
+                  label="Your room name"
+                  error={!!error}
+                  helperText={error?.message}
+                  fullWidth
+                  value={value}
+                  onChange={onChange}
+                  autoFocus
+                />
+                <Button
+                  onClick={generateRandomRoomName}
+                  sx={{ mt: 2 }}
+                  variant="contained"
+                >
+                  Random Name
+                </Button>
+              </>
```

Comment:
Can u add a screenshot of how this looks like in the form? in the description
Also maybe a text field adornment https://mui.com/material-ui/react-text-field/#input-adornments might look better

Reply:
I add a screenshot so you can see what I've done.
I think if I use text field adornment there won't be an icon to symbole ( generate name ), I think a button will be the best choice.

Reply:
okay fair points, but you can use it with a combination of a tooltip, im not a designer im good with this